clang/test/Driver/sycl-oneapi-gpu.cpp

```diff
@@ -0,0 +1,384 @@
+// MACRO_AMD: clang{{.*}} "-triple" "amdgcn-amd-amdhsa"
+// MACRO_AMD: "-D__SYCL_TARGET_AMD_GPU_[[MAC_STR]]__"
+// DEVICE_AMD: clang-offload-wrapper{{.*}} "-compile-opts=--offload-arch=[[DEV_STR]]"
+// MACRO_AMD: clang{{.*}} "-fsycl-is-host"
+// MACRO_AMD: "-D__SYCL_TARGET_AMD_GPU_[[MAC_STR]]__"
+
+/// -fsycl-targets=spir64_x86_64 should set a specific macro
+// RUN: %clangxx -c -fsycl -fsycl-targets=spir64_x86_64 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=MACRO_X86_64
+// RUN: %clang_cl -c -fsycl -fsycl-targets=spir64_x86_64 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=MACRO_X86_64
+// MACRO_X86_64: clang{{.*}} "-triple" "spir64_x86_64-unknown-unknown"
+// MACRO_X86_64: "-D__SYCL_TARGET_INTEL_X86_64__"
+// MACRO_X86_64: clang{{.*}} "-fsycl-is-host"
+// MACRO_X86_64: "-D__SYCL_TARGET_INTEL_X86_64__"
+
+/// test for invalid intel arch
+// RUN: %clangxx -c -fsycl -fsycl-targets=intel_gpu_bad -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=BAD_INPUT
+// RUN: %clang_cl -c -fsycl -fsycl-targets=intel_gpu_bad -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=BAD_INPUT
+// BAD_INPUT: error: SYCL target is invalid: 'intel_gpu_bad'
+
+/// test for invalid nvidia arch
+// RUN: %clangxx -c -fsycl -fsycl-targets=nvidia_gpu_bad -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=BAD_NVIDIA_INPUT
+// RUN: %clang_cl -c -fsycl -fsycl-targets=nvidia_gpu_bad -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=BAD_NVIDIA_INPUT
+// BAD_NVIDIA_INPUT: error: SYCL target is invalid: 'nvidia_gpu_bad'
+
+/// test for invalid amd arch
+// RUN: %clangxx -c -fsycl -fsycl-targets=amd_gpu_bad -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=BAD_AMD_INPUT
+// RUN: %clang_cl -c -fsycl -fsycl-targets=amd_gpu_bad -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=BAD_AMD_INPUT
+// BAD_AMD_INPUT: error: SYCL target is invalid: 'amd_gpu_bad'
+
+/// Test for proper creation of fat object
+// RUN: %clangxx -c -fsycl -fsycl-targets=intel_gpu_skl \
+// RUN:   -target x86_64-unknown-linux-gnu -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=FATO
+// FATO: clang-offload-bundler{{.*}} "-type=o"
+// FATO: "-targets=sycl-spir64_gen-unknown-unknown-skl,host-x86_64-unknown-linux-gnu"
+
+/// Test for proper creation of fat object
+// RUN: %clangxx -c -fsycl -fsycl-targets=nvidia_gpu_sm_50 \
+// RUN:   -target x86_64-unknown-linux-gnu -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=NVIDIA_FATO
+// NVIDIA_FATO: clang-offload-bundler{{.*}} "-type=o"
+// NVIDIA_FATO: "-targets=sycl-nvptx64-nvidia-cuda-sm_50,host-x86_64-unknown-linux-gnu"
+
+/// Test for proper creation of fat object
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx700 \
+// RUN:   -target x86_64-unknown-linux-gnu -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=AMD_FATO
+// AMD_FATO: clang-offload-bundler{{.*}} "-type=o"
+// AMD_FATO: "-targets=host-x86_64-unknown-linux,hipv4-amdgcn-amd-amdhsa--gfx700"
```

Comment:
Should this be `sycl-amdgcn-amd-amdhsa-gfx700`?  The creation isn't matching the consumption.

Reply:
@mmoadeli, any input on this?

Reply:
Thanks for the information!